src/mark.c

```diff
@@ -141,6 +143,13 @@ gboolean r_mark_active(RaucSlot *slot, GError **error)
 	r_slot_status_load(slot);
 	slot_state = slot->status;
 
+	if (r_context()->config->prevent_late_fallback == R_CONFIG_FALLBACK_LOCK_COUNTER) {
```

Comment:
Please add a comment explaining why locking removal happens before set_primary here.

We can set multiple state variables at once atomically, but we don't. This might very well not be a problem, but it's not immediately clear why. A comment should help explaining why a powercut between removal of lock and set_primar below doesn't harm correctness.